examples/vulnerabilities-connector/app/VulnerabilitiesConnector.go

```diff
@@ -0,0 +1,276 @@
+			log.Printf("Error building %s config: %s", provider, err.Error())
+			continue
+		}
+
+		tenantId := fmt.Sprintf("Zenith Systems - %s", provider)
+		configFilePath := fmt.Sprintf("tenant_store_%s.yaml", provider)
+
+		ctx := context.Background()
+		t, err := common.NewTenant(ctx, tenantId, configFilePath, v.Config.SynqlyOrgToken, map[mgmt.CategoryId]*mgmt.CreateIntegrationRequest{
+			mgmt.CategoryIdVulnerabilities: vulnProvider,
+		})
+		if err != nil {
+			log.Printf("Error creating the Tenant for %s: %s", provider, err.Error())
+			continue
+		}
+
+		v.VulnerabilitiesProvider[provider] = vulnProvider
+		v.Tenants[provider] = t
+	}
+
+	v.Logger.Printf("Example initialized successfully")
+}
+
+func (v *VulnerabilitiesConnector) QueryAssets(ctx context.Context, client *engineClient.Client, provider *mgmt.CreateIntegrationRequest) {
+	filter := []*string{}
+
+	if provider.ProviderConfig.Type != string(NUCLEUS) {
+		filter = append(filter, engine.String("device.last_seen_time[gte]2024-09-01T00:00:00Z"))
+	}
+
+	assets, err := client.Vulnerabilities.QueryAssets(ctx, &engine.QueryAssetsRequest{
+		Filter: filter,
+	})
+	if err != nil {
+		log.Printf("Error executing QueryAssets for %s provider: %s", provider, err.Error())
+		return
+	}
+	v.Logger.Printf("Found %d security assets from %s\n", len(assets.Result), provider.ProviderConfig.Type)
+}
+
+func (v *VulnerabilitiesConnector) QueryFindings(ctx context.Context, client *engineClient.Client, provider *mgmt.CreateIntegrationRequest) {
+	filter := []*string{
+		engine.String("severity[in]Critical"),
+	}
+
+	if provider.ProviderConfig.Type != string(NUCLEUS) {
+		filter = append(filter, engine.String("finding.last_seen_time[gte]2024-09-01T00:00:00Z"))
+	}
+
+	if provider.ProviderConfig.Type == string(TENABLE) {
+		filter = append(filter, engine.String("state[in]New"))
+	}
+
+	findings, err := client.Vulnerabilities.QueryFindings(ctx, &engine.QueryFindingsRequest{
+		Filter: filter,
+	})
+	if err != nil {
+		log.Printf("Error executing QueryFindings for %s provider: %s", provider, err.Error())
+		return
+	}
+	v.Logger.Printf("Found %d security findings from %s\n", len(findings.Result), provider.ProviderConfig.Type)
+}
+
+func (v *VulnerabilitiesConnector) QueryScans(ctx context.Context, client *engineClient.Client, provider *mgmt.CreateIntegrationRequest) {
+	var e *engine.NotImplementedError
+
+	assets, err := client.Vulnerabilities.QueryScans(ctx, &engine.QueryScansRequest{})
+	if err != nil && !errors.As(err, &e) {
+		log.Printf("Error executing QueryScans for %s provider: %s", provider, err.Error())
+		return
+	}
+
+	if errors.As(err, &e) {
+		v.Logger.Printf("Query scans not implemented for %s\n", provider.ProviderConfig.Type)
+	} else {
+		v.Logger.Printf("Found %d scans from %s\n", len(assets.Result), provider.ProviderConfig.Type)
+	}
+}
+
+func (v *VulnerabilitiesConnector) buildVulnerabilityProviderConfig(provider VulnerabilitiesConnectorProvider) (*mgmt.CreateIntegrationRequest, error) {
```

Comment:
pass in the viper config object and base the logic here on that

Please sync with @icemancml and use the same logic he is using -- there's an interface that each provider implements and should be simpler in this section of the code using that